hedera-node/src/test/java/com/hedera/services/tokens/HederaTokenStoreTest.java

```diff
@@ -116,6 +116,7 @@
 import static org.mockito.BDDMockito.verify;
 import static org.mockito.BDDMockito.willCallRealMethod;
 import static org.mockito.BDDMockito.willThrow;
+import static org.mockito.Mockito.spy;
 
 @RunWith(JUnitPlatform.class)
 class HederaTokenStoreTest {
```

Comment:
I think that it will be a good idea to write unit tests for the following scenarios:

- On token create -> Treasury Account is added into `knownTreasuries`
- On token update -> Old treasury account is removed from `knownTreasuries`; new treasury account is added into `knownTreasuries`
- On token update + other token exists -> Create 2 tokens with the same treasury. Update the first token to a new treasury. Check that the old treasury is still there because it is a treasury to a token; Check that the new treasury is added to `knownTreasuries`